cluster/images/conformance/run_e2e.sh

```diff
@@ -39,23 +39,36 @@ saveResults() {
 # We get the TERM from kubernetes and handle it gracefully
 trap shutdown TERM
 
-ginkgo_args=(
-    "--focus=${E2E_FOCUS}"
-    "--skip=${E2E_SKIP}"
-    "--noColor=true"
-)
-
+ginkgo_args=""
 if [[ -n ${E2E_DRYRUN:-} ]]; then
     ginkgo_args+=("--dryRun=true")
 fi
 
+GINKGO_NODES=1
 case ${E2E_PARALLEL} in
-    'y'|'Y')           ginkgo_args+=("--nodes=25") ;;
-    [1-9]|[1-9][0-9]*) ginkgo_args+=("--nodes=${E2E_PARALLEL}") ;;
+    'y'|'Y')           GINKGO_NODES=25 ;;
+    [1-9]|[1-9][0-9]*) GINKGO_NODES=${E2E_PARALLEL} ;;
 esac
 
-echo "/usr/local/bin/ginkgo ${ginkgo_args[@]} /usr/local/bin/e2e.test -- --disable-log-dump --repo-root=/kubernetes --provider=\"${E2E_PROVIDER}\" --report-dir=\"${RESULTS_DIR}\" --kubeconfig=\"${KUBECONFIG}\""
-/usr/local/bin/ginkgo "${ginkgo_args[@]}" /usr/local/bin/e2e.test -- --disable-log-dump --repo-root=/kubernetes --provider="${E2E_PROVIDER}" --report-dir="${RESULTS_DIR}" --kubeconfig="${KUBECONFIG}" | tee ${RESULTS_DIR}/e2e.log &
+if [[ "${GINKGO_NODES}" -gt 1 ]]; then
+    E2E_SKIP="\\[Serial\\]|${E2E_SKIP}"
+fi
+
+ginkgo_args+=(
+    "--nodes=${GINKGO_NODES}"
+    "--focus=${E2E_FOCUS}"
+    "--skip=${E2E_SKIP}"
+    "--noColor=true"
+)
+
+NUM_NODES="$(kubectl get nodes \
+    -o=jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.spec.taints}{"\n"}{end}' \
+    | grep -cv "node-role.kubernetes.io/master" \
```

Comment:
I suspect there are other taints you might want (IIRC hybrid windows/linux clusters are supposed to be tainting their windows nodes out) but I won't block for trying this out

Reply:
copy pasted from @BenTheElder 's code.

Reply:
i spoke to sig-windows about hybrid and there is a tracking issue somewhere, but not sure if this will land this cycle. last time i've checked there was a blocker in the e2e framework on this front.

Reply:
@spiffxp `70431`

Reply:
My code only needed to handle kind clusters specifically, so we're depending on kubeadm's behavior + our own. Not sure if this is sufficient for all clusters.

Reply:
i guess that's true. someone might decide to allow scheduling workloads on their CP nodes.
will try to propose something more generic.